netty/src/main/java/io/grpc/netty/GrpcHttp2ConnectionHandler.java

```diff
@@ -34,6 +35,8 @@ public GrpcHttp2ConnectionHandler(Http2ConnectionDecoder decoder,
     super(decoder, encoder, initialSettings);
   }
 
+  protected ChannelPromise channelUnused;
```

Comment:
Could this be initialized from the constructor? Having the super-class reach in is harder to follow. It still being @Nullable is fine. I'd be fine with either two constructors (one with and one without the promise) or having the client pass in a `null` instance.

Reply:
Done.  When I was making this CL, I thought there was a cycle between creating the channel (for the promise) and the handler.  That only ended up in the tests, so Done.